internal/locate/region_cache.go

```diff
@@ -1315,8 +1375,11 @@ func (c *RegionCache) insertRegionToCache(cachedRegion *Region) {
 		if InvalidReason(atomic.LoadInt32((*int32)(&oldRegion.invalidReason))) == NoLeader {
 			store.workTiKVIdx = (oldRegionStore.workTiKVIdx + 1) % AccessIndex(store.accessStoreNum(tiKVOnly))
 		}
-		// Invalidate the old region in case it's not invalidated and some requests try with the stale region information.
-		oldRegion.invalidate(Other)
+		// If the region info is async reloaded, the old region is still valid.
+		if atomic.LoadInt32(&oldRegion.asyncReload) == 0 {
```

Comment:
This logic is a bit strange, why need this judgment?

Reply:
If the region is async reloaded, the old region info is still valid(just , so requests from other threads using the old one should be able to continue instead of failed by invalid region cache.

Reply:
It seems we can't judge if the `region` is async reloaded or not only with the condition `atomic.LoadInt32(&oldRegion.asyncReload) == 0`? The zero value could be either no-op or async reload has been finished.

If our purpose is to skip invalidation if the `insertRegionToCache` is triggered by an `async reload`, how about passing the triggering reason as a function input parameter instead of an atomic variable `asyncReload`?

Reply:
> how about passing the triggering reason as a function input parameter

What do you mean by passing the reason, it can be an enum or a bool value indicates whether invalidate the old one, I'm inclined to make it easy like a paramater `invalidateOldRegion bool` is enough.

Reply:
@you06 
A parameter `invalidateOldRegion` bool is fine by now.